engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -391,6 +391,32 @@ private static void tagCommand(Command cmd) {
         }
     }
 
+    /**
+     * @param commands
+     * @return
+     */
+    private Command[] checkForCommandsAndTag(Commands commands) {
+        Command[] cmds = commands.toCommands();
+
+        assert cmds.length > 0 : "Ask yourself this about a hundred times.  Why am I  sending zero length commands?";
```

Comment:
it was done like this in two places by Alena. I had the idea she had a goal in mind. the assert is only nice in debug situation so her goal was probably educating developers. the following if negates it, kind of completely.

@bhaisaab it is also a test of the new findbugs messages that i am leaving it in for. Shall we?